components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+            IdentityMessageContext messageContext, IdentityRequest identityRequest,
+            JWTInboundResponse.JWTInboundResponseBuilder respBuilder) {
+
+        AuthenticationResult authenticationResult = processResponseFromFrameworkLogin(messageContext,
+                identityRequest);
+        if (log.isDebugEnabled()) {
+            log.debug("Session ID exists.");
+        }
+        if (authenticationResult != null && authenticationResult.isAuthenticated()) {
+            // Authenticated session - response coming from the framework after authentication.
+            respBuilder = handleAuthenticationResult(identityRequest, authenticationResult, respBuilder);
+        } else {
+            // Non-authenticated session - response coming after logged out - redirecting the user to Logout URL
+            respBuilder = handleLogoutResult(identityRequest, respBuilder);
+        }
+        return respBuilder;
+    }
+
+    /**
+     * Handles the response coming from the framework after authentication.
+     *
+     * @param identityRequest      The identity request
+     * @param authenticationResult The authentication result
+     * @param respBuilder          The response builder
+     * @return The response builder after setting the required options
+     */
+    private JWTInboundResponse.JWTInboundResponseBuilder handleAuthenticationResult(
+            IdentityRequest identityRequest, AuthenticationResult authenticationResult,
+            JWTInboundResponse.JWTInboundResponseBuilder respBuilder) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Authenticated session exists. Treating as a request coming after the " +
+                    "authentication.");
+        }
+        String userName = authenticationResult.getSubject().getUserName();
+        if (log.isDebugEnabled()) {
+            log.debug("Processing the request for the user: " + neutralize(userName));
+        }
+
+        // Validate API key and Generate JWT Token
+        Map<ClaimMapping, String> userAttributes = authenticationResult.getSubject().getUserAttributes();
+        String apiKey = getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.API_KEY);
+        if (StringUtils.isNotBlank(apiKey)) {
+            try {
+                respBuilder.setToken(generateJWTToken(identityRequest, apiKey, userName,
+                        userAttributes));
+            } catch (JWTIdentityException e) {
+                String msg = "Error while generating JWT Token";
+                log.error(msg, e);
+                return JWTInboundUtil
+                        .sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                                JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+            }
+        } else {
+            String msg = "Mandatory configuration: API Key is not configured.";
+            log.error(msg, new JWTIdentityException(msg));
```

Comment:
This is something we are checking from the incoming request right? If so make it a debug log

Reply:
A common error will be shown to the client. But if we make it debug, it becomes mandatory to enable debug level logs to understand the actual error on the log. And this configuration should be configured from the WSO2 IS side.